workflows/prognostic_c48_run/runtime/loop.py

```diff
@@ -344,10 +396,36 @@ def _step_dynamics(self) -> Diagnostics:
         # no diagnostics are computed by default
         return {}
 
-    def _compute_physics(self) -> Diagnostics:
-        self._log_debug(f"Physics Step (compute)")
-        self._fv3gfs.compute_physics()
-        # no diagnostics are computed by default
+    def _step_pre_radiation_physics(self) -> Diagnostics:
+        self._log_debug(f"Pre-radiation Physics Step")
+        self._fv3gfs.step_pre_radiation()
+        if self._radiation_wrapper is not None:
+            self._radiation_wrapper.rad_update(self.time, self._timestep)
+        return {
+            f"{name}_pre_radiation": self._state[name]
```

Comment:
magic string here...could be tricky to track down this later on.

Reply:
Thoughts on how to get around this? We're limited by the fact that all diags must be pooled in the `diagnostics` dict over the course of the substeps, so that if we want to get the model states at different substeps separately (the use case here) there's a name clash. It seems like we need handling of a `substep` xarray dimension on the diagnostics which can then be parsed by the diagnostics manager to put it in the right zarr at the end of the model timestep. That's a lot of new diagnostics handling capabilities though, so I temporarily punted and just suffixed the diagnostics, unsatisfying as that is.

Reply:
I'd probably try to move this kind of renaming up the call-stack they way we do with `.e.g self.monitor` in `__iter__`. just feels odd for a function to know its own name sometimes.

Reply:
punting on this. It's a small amount of tech debt for now.